src/channel_state.ts

```diff
@@ -526,14 +555,26 @@ export class ChannelState<
       messageArr = messageArr.filter((msg) => !(msg.id && message.id === msg.id));
     }
 
-    // for empty list just concat and return
-    if (messageArr.length === 0) return messageArr.concat(message);
+    // for empty list just concat and return unless it's an update or deletion
+    if (messageArr.length === 0 && addMessageToList) {
+      return messageArr.concat(message);
+    } else if (messageArr.length === 0) {
+      return [...messageArr];
+    }
 
     const messageTime = (message[sortBy] as Date).getTime();
 
-    // if message is newer than last item in the list concat and return
-    if ((messageArr[messageArr.length - 1][sortBy] as Date).getTime() < messageTime)
+    // if message is newer than last item in the list concat and return unless it's an update or deletion
+    if (
+      (messageArr[messageArr.length - 1][sortBy] as Date).getTime() < messageTime &&
```

Comment:
maybe a separate const for `(messageArr[messageArr.length - 1][sortBy] as Date).getTime()` as well?